pkg/protocol/engine/attestation/attestation.go

```diff
@@ -0,0 +1,20 @@
+package attestation
+
+import (
+	"github.com/iotaledger/hive.go/ads"
+	"github.com/iotaledger/hive.go/runtime/module"
+	"github.com/iotaledger/iota-core/pkg/protocol/engine/blocks"
+	iotago "github.com/iotaledger/iota.go/v4"
+)
+
+type Attestation interface {
+	Get(index iotago.SlotIndex) (attestations *ads.Map[iotago.AccountID, iotago.Attestation, *iotago.AccountID, *iotago.Attestation], err error)
+	AddAttestationFromBlock(block *blocks.Block)
+	Commit(index iotago.SlotIndex) (newCW uint64, attestationsMap *ads.Map[iotago.AccountID, iotago.Attestation, *iotago.AccountID, *iotago.Attestation], err error)
+
+	// TODO: integrate into snapshot
+	//  Import(reader io.ReadSeeker) (err error)
+	//  Export(writer io.WriteSeeker, targetSlot iotago.SlotIndex) (err error)
```

Comment:
That's not correct: we can create a snapshot up to the latest commitment index. A node needs to be able to bootstrap and continue running from there. Since we consider slots from before the last commitment when creating a new commitment, we need that information. Specifically, I think we need to export (and persist on disk for correct restoring of a node after restart) the attestations that are in the tracker. So that we can compute the correct root and CW when creating a new commitment.